examples/newPlacesNearbySearch/example.js

```diff
@@ -0,0 +1,133 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// This example showcases the New Place nearbySearch API, as well as drawing a circle on a Google Map.
+// The example draws a circle on the map, for which the radius can be scaled dynamically.
+// The user can choose from different place type categories, and then perform a nearbySearch,
+// which will return any places matching the specified type within the circle radius.
+// This example also makes use of dynamic imports and advanced markers support.
+//
+// This is meant to be showcased as the client logic that is able to remain untouched
+// and retain the same functionality when using the Amazon Location Migration SDK.
+
+let map;
```

Comment:
I don't love having dynamic javascript variables that are declared outside of a function or component to own them. It's fine for constants.  With the multiple imports or rerenders of a js file sometimes things can get a bit weird. I'm thinking more react/framework which this isn't so it might be overkill. But service isn't consumed anywhere and could be removed, markers and searchCircle is only used in one function so that function could own them.  Maps is used in both, but it seems like initMap could either take it as parameter or return it depending on your flow so it's constructed for the search function.

Reply:
Good call-out about `service`, I will remove.

For the others, in practice you wouldn't want to design your application like this, you are correct. I only have it like this because these are demo examples, so having these global variables like this allow me to be able to test/demo different functionality on the fly (e.g. take the `map` and change the zoom, or move its center, take the `searchCircle` and show how I can change it's color)

Reply:
These are straight html and javascript examples so it's probably fine that their barebone like this. We can show best practices in more full component examples elsewhere.